src/wasm_opt.rs

```diff
@@ -0,0 +1,95 @@
+use anyhow::{anyhow, ensure, Context, Result};
+use lazy_static::lazy_static;
+use std::{
+    fs,
+    path::{Path, PathBuf},
+    process,
+};
+
+fn wasm_opt_url() -> &'static str {
+    lazy_static! {
+        static ref WASM_OPT_URL: String = {
+            format!(
+                "https://github.com/WebAssembly/binaryen/releases/download/version_{version}/binaryen-version_{version}-{arch}-{os}.tar.gz",
+                version = "105",
+                arch = platforms::TARGET_ARCH,
+                os = platforms::TARGET_OS,
+            )
+        };
+    }
+
+    &WASM_OPT_URL
+}
+
+pub fn run(
+    binary_path: impl AsRef<Path>,
+    shrink_level: u32,
+    optimization_level: u32,
+    debug_info: bool,
+) -> Result<()> {
+    let wasm_opt = download_wasm_opt()?;
+    let output_path = binary_path.as_ref().with_extension("opt");
+
+    let mut command = process::Command::new(&wasm_opt);
+    command
+        .stderr(process::Stdio::inherit())
+        .arg(binary_path.as_ref())
+        .args(&[
+            "-o",
+            &output_path.to_str().expect("cannot parse output path"),
+            "-O",
+        ])
+        .args(&["-ol", &optimization_level.to_string()])
+        .args(&["-s", &shrink_level.to_string()]);
```

Comment:
That's why `.arg()` takes `impl AsRef<OsStr>` :tada: 
```suggestion
        .arg("-ol")
        .arg(optimization_level.to_string())
        .arg("-s")
        .arg(shrink_level.to_string())
```